framework/Models/PostProcessors/PostProcessorInterface.py

```diff
@@ -0,0 +1,134 @@
+    self.printTag = 'PostProcessorInterface'
+    self.outputDataset  = False # True if the user wants to dump the outputs to dataset
+    self.validDataType = ['PointSet','HistorySet'] # The list of accepted types of DataObject
+    ## Currently, we have used both DataObject.addRealization and DataObject.load to
+    ## collect the PostProcessor returned outputs. DataObject.addRealization is used to
+    ## collect single realization, while DataObject.load is used to collect multiple realizations
+    ## However, the DataObject.load can not be directly used to collect single realization
+    ## One possible solution is all postpocessors return a list of realizations, and we only
+    ## use addRealization method to add the collections into the DataObjects
+    self.outputMultipleRealizations = False
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, it is the run info from the jobHandler
+      @ In, inputs, list, it is a list containing whatever is passed with an input role in the step
+      @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
+      @ Out, None
+    """
+    super().initialize()
+    self.inputs = inputs
+    if 'stepName' in runInfo:
+      self.workingDir = os.path.join(runInfo['WorkingDir'],runInfo['stepName']) #generate current working dir
+    else:
+      self.workingDir = runInfo['WorkingDir']
+
+  @abc.abstractmethod
+  def run(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process.
+      Should avoid to use (inputToInternal output), and passing xarray directly/dataset
+      Possible inputs include: dict, xarray.Dataset, pd.DataFrame
+      @ Out, dict, xarray.Dataset, pd.DataFrame --> I think we can avoid collectoutput in the plugin pp
+    """
+
+  def collectOutput(self,finishedJob, output, options=None):
+    """
+      Method that collects the outputs from the "run" method of the PostProcessor
+      @ In, finishedJob, InternalRunner object, instance of the run just finished
+      @ In, output, "DataObjects" object, output where the results of the calculation needs to be stored
+      @ In, options, dict, optional, not used in PostProcessor.
+        dictionary of options that can be passed in when the collect of the output is performed by another model (e.g. EnsembleModel)
+      @ Out, None
+    """
+    if output.type not in self.validDataType:
+      self.raiseAnError(IOError, 'Output type', str(output.type), 'is not allowed!')
+    evaluation = finishedJob.getEvaluation()
+    if isinstance(evaluation, Runners.Error):
```

Comment:
please open an issue. This "error" must be handled at the JobHandler level. We should not reach this point.

Reply:
see #1524